drivers/sensor/nxp/fxls8974/fxls8974.c

```diff
@@ -0,0 +1,632 @@
+/*
+ * Copyright 2024 NXP
+ * Copyright (c) 2018 Phytec Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_fxls8974
+
+#include "fxls8974.h"
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+#include <stdlib.h>
+
+LOG_MODULE_REGISTER(FXLS8974, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#define DIR_READ(a)  ((a) & 0x7f)
+#define DIR_WRITE(a) ((a) | BIT(7))
+#define ADDR_7(a) ((a) & BIT(7))
+
+int fxls8974_transceive(const struct device *dev,
+				void *data, size_t length)
+{
+		const struct fxls8974_config *cfg = dev->config;
+		const struct spi_buf buf = { .buf = data, .len = length };
+		const struct spi_buf_set s = { .bufs = &buf, .count = 1 };
+
+		return spi_transceive_dt(&cfg->bus_cfg.spi, &s, &s);
+}
+
+int fxls8974_read_spi(const struct device *dev,
+				uint8_t reg,
+				void *data,
+				size_t length)
+{
+		const struct fxls8974_config *cfg = dev->config;
+
+		/* Reads must clock out a dummy byte after sending the address. */
+		uint8_t reg_buf[3] = { DIR_READ(reg), ADDR_7(reg), 0 };
+		const struct spi_buf buf[2] = {
+				{ .buf = reg_buf, .len = 3 },
+				{ .buf = data, .len = length }
+		};
+		const struct spi_buf_set tx = { .bufs = buf, .count = 1 };
+		const struct spi_buf_set rx = { .bufs = buf, .count = 2 };
+
+		return spi_transceive_dt(&cfg->bus_cfg.spi, &tx, &rx);
+}
+
+int fxls8974_byte_read_spi(const struct device *dev,
+				uint8_t reg,
+				uint8_t *byte)
+{
+		/* Reads must clock out a dummy byte after sending the address. */
+		uint8_t data[] = { DIR_READ(reg), ADDR_7(reg), 0};
+		int ret;
+
+		ret = fxls8974_transceive(dev, data, sizeof(data));
+
+		*byte = data[2];
+
+		return ret;
+}
+
+int fxls8974_byte_write_spi(const struct device *dev,
+				uint8_t reg,
+				uint8_t byte)
+{
+		uint8_t data[] = { DIR_WRITE(reg), ADDR_7(reg), byte };
+
+		return fxls8974_transceive(dev, data, sizeof(data));
+}
+
+int fxls8974_reg_field_update_spi(const struct device *dev,
+				uint8_t reg,
+				uint8_t mask,
+				uint8_t val)
+{
+		uint8_t old_val;
+
+		if (fxls8974_byte_read_spi(dev, reg, &old_val) < 0)
+			return -EIO;
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style